### PR TITLE
terminal: fix 'xterm' and addon dependencies to support searching

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: #9164

The following commit updates the `xterm` addon dependencies so that they are compatible with the `xterm` version.
The changes including pinning the xterm version as to not result in future incompatibilities and updating the addon
versions to match.

- [x] confirmed that the updated dependencies are license compatible using our 'clearly defined' process.

<div align='center'>

https://user-images.githubusercontent.com/40359487/110523625-5f031800-80e0-11eb-908d-2843c0b095e7.mov

</div>

#### How to test

1. start the application, and spawn a terminal
2. produce output (ex: `ls -al`)
3. perform <kbd>ctrlcmd</kbd>+<kbd>f</kbd>
4. include a search input and confirm it works, confirm that the arrows (previous and next) work

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by marechal-p** [comment]
Breakage appeared in `xtermjs@4.5.0`: https://github.com/xtermjs/xterm.js/commit/44ff7b0f5f6c7ff4e3e081269b2b88587326f597#diff-d060abc72ee9e32f474e8bb163d0e8e5b9373030b4fb52c9c32ca08b91f446c9L577

File: packages/terminal/package.json [hunk 1]
```diff
@@ -8,9 +8,9 @@
     "@theia/filesystem": "1.11.0",
     "@theia/process": "1.11.0",
     "@theia/workspace": "1.11.0",
-    "xterm": "^4.4.0",
-    "xterm-addon-fit": "^0.3.0",
-    "xterm-addon-search": "^0.5.0"
+    "xterm": "~4.4.0",
+    "xterm-addon-fit": "~0.3.0",
+    "xterm-addon-search": "~0.5.0"
```
Comment (marechal-p) [inline packages/terminal/package.json:13]: We should be able to leave those untouched for now.

**Comment by vince-fugnitto** [comment]
I've updated the pull-request to bump the version of `xterm` but target a more specific version as to not have future incompatibility issues.

**Review by marechal-p — APPROVED** [review]
LGTM

**Pull request merged**